app/javascript/main.jsx

```diff
@@ -28,63 +28,74 @@ import EditUser from './components/admin/manage_users/EditUser';
 import EditRole from './components/admin/roles/EditRole';
 import Home from './components/home/Home';
 import ActivateAccount from './components/users/account_activation/ActivateAccount';
-import ErrorBoundary from './components/shared_components/ErrorBoundary';
-import DefaultErrorPage from './components/errors/DefaultErrorPage';
-import NotFoundPage from './components/errors/NotFoundPage';
 import VerifyAccount from './components/users/account_activation/VerifyAccount';
 import AdminPanel from './components/admin/AdminPanel';
 import UnauthenticatedOnly from './routes/UnauthenticatedOnly';
 import AuthenticatedOnly from './routes/AuthenticatedOnly';
+import DefaultErrorPage from './components/errors/DefaultErrorPage';
+import NotFoundPage from './components/errors/NotFoundPage';
 
-const queryClient = new QueryClient();
+const queryClientConfig = {
+  defaultOptions: {
+    queries: {
+      useErrorBoundary: true,
+    },
+  },
+};
 
```

Comment:
@scouillard I think this was the only missing piece. I tested this and it works as expected